pkg/services/ngalert/api/generated_base_api_provisioning.go

```diff
@@ -100,23 +100,23 @@ func (f *ProvisioningApiHandler) RoutePostAlertRule(ctx *models.ReqContext) resp
 	// Parse Request Body
 	conf := apimodels.ProvisionedAlertRule{}
 	if err := web.Bind(ctx.Req, &conf); err != nil {
-		return response.Error(http.StatusBadRequest, "bad request data", err)
+		return response.Error(http.StatusBadRequest, err.Error(), err)
```

Comment:
So the API already does return this error string, just in a separate field.

For example, when I force a deserialization error, I get something like:
```
{
  "error": "json: cannot unmarshal number into Go struct field EmbeddedContactPoint.type of type string",
  "message": "bad request data",
  "traceID": ""
}
```
The string we're looking for is already being returned, and adding it here would duplicate it. Did we see a place where the `error` string was not being correctly displayed to the user? If so, that feels like a bug in the presentation layer and not here.

Reply:
According to #56387 the string is being returned when running in non-production mode due to this [condition ](https://github.com/grafana/grafana/blob/53baecd71f5519e0ef300f3067cb89eab2908e2f/pkg/api/response/response.go#L213) .
The fix here should provide a proper error string under the `message` field when running in production mode.

Reply:
Wow, that's unfortunate that there is such a difference between debug and release builds. Nice find!